plugins/c8y_log_plugin/src/main.rs

```diff
@@ -134,16 +120,13 @@ async fn run(
                     return Ok(());
                 }
             }
-            Some(event_or_error) = inotify_stream.next() => {
-                if let Ok(event) = event_or_error {
-                    match event.mask {
-                        EventMask::CLOSE_WRITE => {
-                            plugin_config = handle_dynamic_log_type_update(mqtt_client, config_file).await?;
-                        }
-                        _ => {}
+            Some(Ok(event)) = inotify_stream.next() => {
+                match event.mask {
+                    EventMask::CLOSE_WRITE => {
+                        plugin_config = handle_dynamic_log_type_update(mqtt_client, config_file).await?;
                     }
+                    _ => {}
                 }
```

Comment:
This is were you idea of adding a `filter_map` to the `inotify` stream can really shine.

1) To avoid this cascade of matches, first on ` Some(Ok(event))` and then on the `event.mask`.
2) To avoid this C-like enums: `EventMask::CLOSE_WRITE` and `WatchMask::CLOSE_WRITE`
3) And why not to use a single enum for `EventMask` and `WatchMask`.